components/tikv_util/src/future_pool/builder.rs

```diff
@@ -76,4 +111,14 @@ impl Builder {
         let pool = Arc::new(self.inner_builder.build());
         super::FuturePool { pool, env }
     }
+
+    /// Builds the FuturePool which limits max tasks.
+    pub fn build_with_task_limit(&mut self) -> super::TaskLimitedFuturePool {
```

Comment:
why not let the FuturePool support limiting task directly? I think we can check task limited in FuturePool spawn directly.

Reply:
I marked a TODO for it. This looks to be pretty easy for contributors to change. Currently `FuturePool` is also used in somewhere, which provides different signatures compared to `TaskLimitedFuturePool`. More specifically, `FuturePool` provides non-failure interfaces while `TaskLimitedFuturePool` provides interfaces wrapped by `Result`.

Reply:
I prefer you providing FuturePool at first. 

Which mean first supporting task limited for the FuturePool (you may need to handle result of spawn too), then remove the ReadPool.

Reply:
/cc @BusyJay

Reply:
Supporting limited task first makes sense to me.